src/sources/opentelemetry/log.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{
+    config::{
+        AcknowledgementsConfig, DataType, GenerateConfig, Output, Resource, SourceConfig,
+        SourceContext,
+    },
+    internal_events::{EventsReceived, StreamClosedError},
+    serde::bool_or_struct,
+    sources::{util::grpc::run_grpc_server, Source},
+    tls::{MaybeTlsSettings, TlsEnableableConfig},
+    SourceSender,
+};
+use futures::TryFutureExt;
+use otel_proto::LogService::{
+    logs_service_server::{LogsService, LogsServiceServer},
+    ExportLogsServiceRequest, ExportLogsServiceResponse,
+};
+use serde::{Deserialize, Serialize};
+use std::net::SocketAddr;
+use tonic::{Request, Response, Status};
+use vector_core::{
+    event::{BatchNotifier, BatchStatus, BatchStatusReceiver, Event},
+    ByteSizeOf,
+};
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct OpentelemetryLogConfig {
+    address: SocketAddr,
+    #[serde(default)]
+    tls: Option<TlsEnableableConfig>,
+    #[serde(default, deserialize_with = "bool_or_struct")]
+    acknowledgements: AcknowledgementsConfig,
+}
+
+impl GenerateConfig for OpentelemetryLogConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(Self {
+            address: "0.0.0.0:6788".parse().unwrap(),
+            tls: None,
+            acknowledgements: Default::default(),
+        })
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "opentelemetry")]
+impl SourceConfig for OpentelemetryLogConfig {
+    async fn build(&self, cx: SourceContext) -> crate::Result<Source> {
+        let tls_settings = MaybeTlsSettings::from_config(&self.tls, true)?;
+        let acknowledgements = cx.do_acknowledgements(&self.acknowledgements);
+        let service = LogsServiceServer::new(Service {
+            pipeline: cx.out,
+            acknowledgements,
+        })
+        .accept_gzip();
+        let source =
+            run_grpc_server(self.address, tls_settings, service, cx.shutdown).map_err(|error| {
+                error!(message = "Source future failed.", %error);
+            });
+
+        Ok(Box::pin(source))
+    }
+
+    fn outputs(&self) -> Vec<Output> {
+        vec![Output::default(DataType::Log)]
+    }
```

Comment:
As the opentelemetry source will likely be extended to also support metrics in the near future, we should follow the same convention as [this source](https://github.com/vectordotdev/vector/blob/master/src/sources/datadog/agent/mod.rs#L201-L211).